### ISSUE TITLE
Java classes, that have static nested classes ("Unable to process is")

### ISSUE THREAD

**Issue by Coolmanakov** [google/ksp#1086]
I migrate kotlin version of project from 1.6 to 1.7.1.
That's my problem after bumping version of kotlin and ksp, I've faced with [ksp] Unable to process:......MyProcessorName: MyElements.
I can't share code with you, but I can say, that after decrease kotlin and ksp versions, everything process successfully 
Here example of mine:

                  val elements = resolver
                              .getSymbolsWithAnnotation(annotationName)
                              .filterIsInstance<KSClassDeclaration>()
                              
                  elements.filter { it.validate() }.mapNotNull {
                              it.accept(
                                  myVisitor,
                                  data
                              )
                          }.also {output.add(it.toList() }
                  return elements { it.validate() }.toList()

Everything works as expected, except for classes from Java that have static nested classes - for them (in version 1.7) the warning [ksp] "Unable to process is:"

**Comment by Coolmanakov** [comment]
I try to fin answer inside the library
1. I find that java classes with static nested asStarProjectedType() doesn't work for them (isError)

**Comment by neetopia** [comment]
can you try out out out snapshot build that contains a fix to a similar issue to your description? 

https://kotlinlang.slack.com/archives/C013BA8EQSE/p1660942428894469?thread_ts=1657235393.876469&cid=C013BA8EQSE

**Comment by Coolmanakov** [comment]
I don't know why, but I can't open link, some problems with access. But it will be a great pleasure if you describe or share solution here)

**Comment by neetopia** [comment]
it is on kotlin slack channel. 

here is copy of the text
```
While the fix is already merged in main branch, before our next release, 
we decided to release a snapshot that contains the fix for you to verify if our fix is working for your project. 
you can add
maven("https://oss.sonatype.org/content/repositories/snapshots/")
to your build script and use 1.7.20-Beta-1.0.7-SNAPSHOT version of KSP and 1.7.20-Beta compiler to verify.

```

**Comment by Coolmanakov** [comment]
I add maven("https://oss.sonatype.org/content/repositories/snapshots/") to build.gradle and I get "Directory listening forbidden"
I've tried to do it with and without vpn, result's are same
:(

**Comment by Coolmanakov** [comment]
I've updated, but nothing has changed

**Comment by neetopia** [comment]
there might be some issue with snapshot repository, if you open https://oss.sonatype.org/content/repositories/snapshots/com/google/devtools/ksp/ you can find our snapshot build there.

**Comment by Coolmanakov** [comment]
I found that for a nested class in java it returns true, but for a java class that contains a static class among syperTypes, none can be resolve (), because NPE java.lang.NullPointerException: null cannot be cast to non-null type org.jetbrains.kotlin.descriptors.ClassDescriptor
at com.google.devtools.ksp.processing.impl.ResolverImpl.resolveJavaType(ResolverImpl.kt:656)
at com.google.devtools.ksp.processing.impl.ResolverImpl.resolveUserType(ResolverImpl.kt:739)
version is 1.7.20-Beta-1.0.7-SNAPSHOT

**Comment by Coolmanakov** [comment]
@neetopia Are there any solutions for my issue? Looks like it's a bug, 1.7.20-Beta-1.0.7-SNAPSHOT didn't fix the situation.

**Comment by neetopia** [comment]
can you provide a reproduce repo? I tried to add following test case to our `validateTypes` test, and the test result returns `valid` for me.

```
// FILE: StaticMember.java
public class StaticMember {
    static class Member {}
}
```

**Comment by Coolmanakov** [comment]
@neetopia  https://github.com/Coolmanakov/KSP_java_nested_class/tree/master
 I did example repo, check by link

**Comment by Coolmanakov** [comment]
@neetopia 
I debugged the library in search of a problem, but I didn’t find anything concrete, maybe my attempts will give you some idea.
For a static class in the JavaDescriptorResolver class, the resolveClass method returns null
javaClass.outerClass?.let { outerClass ->
            val outerClassScope = resolveClass(outerClass)?.unsubstitutedInnerClassesScope
            return outerClassScope?.getContributedClassifier(javaClass.name, NoLookupLocation.FROM_JAVA_LOADER) as? ClassDescriptor
        }
further there is no point in clarifying something, I hardly understand the meaning of what is happening further. Maybe this piece of code will give you an idea what could be the problem. But again, this piece of code refers to package org.jetbrains.kotlin.resolve.jvm, not xp, maybe I need to report to org.jetbrains.kotlin?
But perhaps the javaClass instance that is returned from ksp is not valid. It's hard for me to understand this right now.

**Comment by neetopia** [comment]
Thanks for the investigation, I am currently focusing on 1.0.7 milestone issues, will reach back after that.

**Comment by Coolmanakov** [comment]
@neetopia ok, cool. Will wait for an answer.

**Comment by jbarr21** [comment]
We are also seeing this issue in our Motif KSP artifact (https://github.com/uber/motif/issues/249). We have had to revert back to using KAPT since we have many mixed module sources and heavily make use of nested classes. The linked issue provides info, a stacktrace, and a sample project on Kotlin/KSP 1.6 and 1.7 that demonstrate the working and broken cases.

**Comment by neetopia** [comment]
we had a tentative fix for this issue, unfortunately it is not covering all cases, which led to a fix that covers more use cases after 1.0.7 was released. 

Cherry picked 15258e6886fdfc78bd22e03b8deb3dc6101d7a97 to 1.0.7 release branch and verified it is working with motif

**Comment by tyvsmith** [comment]
Thanks for the quick response!

**Comment by Coolmanakov** [comment]
Sounds great, but I'd like to check problem has gone. Maybe you can post snapshot with fix?
Another solution: I've posted a repo (https://github.com/Coolmanakov/KSP_java_nested_class/tree/master), maybe, you can check that commit covers my issue too?

**Comment by neetopia** [comment]
after applying the same local build, I got
```
$ ./gradlew :app:kspDebugKotlin

> Task :app:kspDebugKotlin FAILED
e: [ksp] Everything is good, validate all
e: Error occurred in KSP, check log for detail

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:kspDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
```
and it seems that this means the patch is working for you? Can you confirm?

**Comment by Coolmanakov** [comment]
Yep, that means, that all files was validated by ksp. I very happy of that fact. Thanks for help)))

**Comment by neetopia** [comment]
fixed by #1150
